uaclient/util.py

```diff
@@ -497,6 +497,54 @@ def configure_web_proxy(
         request.install_opener(opener)
 
 
+def configure_snap_proxy(
+    http_proxy: "Optional[str]", https_proxy: "Optional[str]"
+) -> None:
+    """
+    Configure snap to use http and https proxies.
+
+    :param http_proxy: http proxy to be used by snap. If None, it will
+                       not be configured
+    :param https_proxy: https proxy to be used by snap. If None, it will
+                        not be configured
+    """
+    if http_proxy:
+        subp(["snap", "set", "system", "proxy.http={}".format(http_proxy)])
```

Comment:
In uaclient/entitlements.livepatch we use retry_sleeps for any snap cmdline interaction         `retry_sleeps=SNAP_INSTALL_RETRIES`. 

Maybe we add a retry_sleeps param here and pass that into subp. (I'd suggest that we use SNAP_INSTALL_RETRIES value here)
The reason I suggest this is a strong suspicion that we are going to hit race conditions on Azure and AWS waiting for for snap seeding to complete which prompted us to add the [snap install livepatch retries](https://github.com/canonical/ubuntu-advantage-client/blob/main/uaclient/entitlements/livepatch.py#L114-L118)